vespa-feed-client/src/main/java/ai/vespa/feed/client/impl/ApacheCluster.java

```diff
@@ -132,6 +136,8 @@ private static CloseableHttpAsyncClient createHttpClient(FeedClientBuilderImpl b
             throw new IllegalStateException("No adequate SSL cipher suites supported by the JVM");
 
         ClientTlsStrategyBuilder tlsStrategyBuilder = ClientTlsStrategyBuilder.create()
+                                                                              .setTlsDetailsFactory(sslEngine ->
+                                                                                                    new TlsDetails(sslEngine.getSession(), sslEngine.getApplicationProtocol()))
```

Comment:
Didn't we consider this, @bjorncs ? I seem to remember that, but maybe I'm just wrong?